src/main.zig

```diff
@@ -132,6 +134,19 @@ const Instance = struct {
 
         return null;
     }
+
+    fn dump_physical_device_properties(self: Self, physical_devices: []const c.VkPhysicalDevice, writer: anytype) !void {
```

Comment:
can the writer be more specific than anytype?

Reply:
also what about providing some sort of iterator instead that would return all physical devices, so the caller is free to do whatever he wants with it like dumping it. It could also be just an allocated array, but I think an iterator would save us from unnecessary allocations in this case.

Reply:
> can the writer be more specific than anytype?

not if you want to allow users to provide any writer. there is no good way to specify writer types in zig.  anytype is the way its done the zig std lib.